shacl12-core/index.html

```diff
@@ -2232,20 +2193,35 @@ <h4>Severity (sh:resultSeverity)</h4>
 			<section id="value-nodes">
 				<h3>Value Nodes</h3>
 				<p>
-					The <a>validators</a> of most constraint components use the concept of <dfn data-lt="value node">value nodes</dfn>, which is defined as follows:
+					The <a>validators</a> of most constraint components use the concept of <dfn data-lt="value node">value nodes</dfn>, which is defined by the following two sub-sections.
 				</p>
-				<ul>
-					<li>
+				<section id="value-nodes-node-shapes">
+					<h4>Value Nodes of Node Shapes</h4>
+					<p>
 						For <a>node shapes</a> the <a>value nodes</a> are the individual <a>focus nodes</a>, forming a set with exactly one member.
-					</li>
-					<li>
+					</p>
+				</section>
+				<section id="value-nodes-property-shapes">
+					<h4>Value Nodes of Property Shapes</h4>
+					<p>
 						For <a>property shapes</a> with a <a>value</a> for <code>sh:path</code> <code>p</code> the
-						<a>value nodes</a> are the set of <a>nodes</a> in the <a>data graph</a>
+						<a>value nodes</a> are defined as follows.
+					</p>
+					<p>
+						If <code>p</code> is a <a>SHACL property path</a> the value nodes are the set of <a>nodes</a> in the <a>data graph</a>
 						that can be reached from the <a>focus node</a> with the <a>path mapping</a> of <code>p</code>.
-						Unless stated otherwise, the value of <code>sh:resultPath</code> of each validation result is a
-						<a>SHACL property path</a> that <a href="#property-paths">represents</a> an <a>equivalent path</a> to the one provided in the shape.
-					</li>
-				</ul>
+					</p>
+					<p>
+						If <code>p</code> is a <a>node expression</a> the value nodes are the set of <a>output nodes</a> of <code>p</code>
+						when evaluated with the given <a>focus node</a> and the <a>data graph</a> as <a>active graph</a>.
+					</p>
+					<p>
+						Unless stated otherwise, the value of <code>sh:resultPath</code> of each validation result is a "deep copy"
+						of <code>p</code> and any <a>triples</a> that can be reached by transitively traversing the <a>blank nodes</a>
+						that appear in the <a>object</a> position of these triples,
+						see the <a href="https://www.w3.org/submissions/CBD/">Consise Bounded Description</a>.
```

Comment:
I don't understand what this means.

(If this is an attempt to include all the nodes in a list, then this text covers a wider range of possibilities)

Reply:
Do you have an example where this text makes a difference from SHACL 1.0?

Reply:
> For property paths the requirement was already like this.

Kinda. It says that `sh:resultPath` must point at a well-formed SHACL property path. Easy to overlook and even the original shacl-js library only copied a single blank node without the rest of the path.

It's good to spell it out explicitly

Reply:
Thanks, I missed that place further down. Fixed: https://github.com/w3c/data-shapes/commit/e508b68152fa988fcbcaa0850c4e279b218a470d

In order to keep track, I'll mark this discussion as resolved - please reopen if you disagree.